app/routes/datasets/queries.py

```diff
@@ -400,9 +399,8 @@ async def _query_raster_lambda(
     except httpx.TimeoutException:
         raise HTTPException(500, "Query took too long to process.")
 
-    try:
-        response_data = response.json()["body"]
-    except (JSONDecodeError, KeyError):
+    response_data = response.json()["body"]
+    if response_data["status"] != "success":
```

Comment:
I would use `.get` here in case the response has something unexpected. Will Lambda functions always have a body object or can it also be something different? If yes, you might want to catch this as well.

```suggestion
    response_data = response.json()["body"]
    if response_data.get("status") != "success":
```

Reply:
@thomas-maschler I think it'll always have a body if the lambda returns successfully, but you're right that I'm not catching if the lambda returns a general error (i.e. couldn't run the lambda for whatever reason). So I added a check of the status code from the response, and just generally check the body has a status and a data field otherwise it fails.